file/disk.lisp

```diff
@@ -3,31 +3,93 @@
 
 (defpackage :mezzano.disk-file-system
   (:use :cl)
+  (:local-nicknames (:sup :mezzano.supervisor))
   (:export #:read-sector
-           #:write-sector))
+           #:write-sector
+           #:block-device-read
+           #:block-device-write
+           #:block-device-flush))
 
 (in-package :mezzano.disk-file-system)
 
+(defgeneric block-device-read (device lba n-sectors buffer &optional offset))
+
+(defmethod block-device-read
+    ((disk sup::disk) lba n-sectors buffer &optional (start-offset 0))
```

Comment:
The disk symbol can be exported from the supervisor package, along with disk-flush